### PR TITLE
Ensure changes in optimizePackageImports invalidate webpack cache

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Makes sure changes to `optimizePackageImports` apply when there is an existing cache.

Closes NEXT-2817

### CODE REVIEW

**Comment by timneutkens** [comment]
* **#63284** <a href="https://app.graphite.dev/github/pr/vercel/next.js/63284?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by huozhi — APPROVED** [review]

File: packages/next/src/build/webpack-config.ts [hunk 1]
```diff
@@ -2037,6 +2037,7 @@ export default async function getBaseWebpackConfig(
   }
 
   const configVars = JSON.stringify({
+    optimizePackageImports: config?.experimental?.optimizePackageImports,
```
Comment (huozhi) [inline packages/next/src/build/webpack-config.ts:2040]: probably also add other bundling related configs like `transpilePackages` as well?

Comment (timneutkens) [reply packages/next/src/build/webpack-config.ts:2040]: Those are different as they don't affect compilation in this case, i.e. that affects `externals` output which would invalidate the cache.

**Review by shuding — APPROVED** [review]

**Review by sokra — APPROVED** [review]

**Review by XoL1507 — COMMENTED** [review]
*

**Pull request merged**